### PR TITLE
Update the module list sorted by functionality

### LINKED ISSUE #440: 完善按功能分类的GMT模块列表
最早提出于 https://github.com/gmt-china/GMT_docs/pull/439#discussion_r654894233

> Do we want to update the following tables based on https://docs.generic-mapping-tools.org/latest/modules.html#session-management in another PR later?

我们需要重新整理一下 [GMT 模块（按功能分类）](https://docs.gmt-china.org/latest/module/) 列表，这里包含的模块也不全。

- [x] main program: #495

### PR DESCRIPTION
**Pull request by core-man**
To make this PR friendly to be reviewed, I'd like to split it into several rounds of reviews. I.e., only a few modules are updated each time during one round of review.

- [GMT 模块](https://docs.gmt-china.org/latest/module/) 
- [GMT modules](https://docs.generic-mapping-tools.org/latest/modules.html#session-management)

**Preview:** https://gmt-china.github.io/sitepreview/gmt-china/GMT_docs/module-table/module/

Fixes #440.

### CODE REVIEW

File: source/module/index.rst [hunk 1]
```diff
@@ -175,10 +175,14 @@
 +-----------------------------------------------+-----------------------------------------------------------------------+
 | **主程序/脚本**                               |                                                                       |
 +===============================================+=======================================================================+
-| :doc:`gmt`                                    | GMT主程序                                                             |
+| :doc:`gmt`                                    | GMT 主程序                                                            |
 +-----------------------------------------------+-----------------------------------------------------------------------+
 | :doc:`gmt-config`                             | 返回 GMT 动态函数库的基本信息                                         |
 +-----------------------------------------------+-----------------------------------------------------------------------+
+| :doc:`gmt:gmt5syntax`                         | 将旧版本的 GMT 脚本转换 ``gmt <module>`` 语法                         |
```
Comment (core-man) [inline source/module/index.rst:182]: See https://docs.generic-mapping-tools.org/latest/gmt5syntax.html:

> # gmt5syntax
> Convert old GMT script to use new ‘gmt <module>’ syntax

File: source/module/index.rst [hunk 2]
```diff
@@ -175,10 +175,14 @@
...
++-----------------------------------------------+-----------------------------------------------------------------------+
+| :doc:`gmt:gmt_shell_functions.sh`             | 在 Bash 脚本中支持 GMT 的实用函数                                     |
```
Comment (core-man) [inline source/module/index.rst:184]: See https://docs.generic-mapping-tools.org/latest/gmt_shell_functions.sh.html:
> ## gmt_shell_functions.sh
> gmt_shell_functions.sh - Practical functions to be used in GMT Bourne Again shell scripts

> ## Description
> gmt_shell_functions.sh provides a set of functions to Bourne (again) shell scripts in support of GMT. The calling shell script should include the following line, before the functions can be used:

File: source/module/index.rst [hunk 3]
```diff
@@ -175,10 +175,14 @@
 +-----------------------------------------------+-----------------------------------------------------------------------+
 | **主程序/脚本**                               |                                                                       |
```
Comment (core-man) [inline source/module/index.rst:176]: [中文手册模块列表](https://docs.gmt-china.org/latest/module/)：
- 将 `gmtswhich` 单独列在 **其他脚本** 分类中
- 将 `img2grd` 列在 **其他模块** 分类中

[GMT 官方模块列表]中二者均在[Utility Scripts](https://docs.generic-mapping-tools.org/latest/modules.html#utility-scripts)中，没有放入按功能分类列表。

File: source/module/index.rst (hunk 1 above)
Comment (seisman) [reply source/module/index.rst:182]: 当初不加 gmt5syntax 和 gmt_shell_functions.sh 是因为觉得这两个没用

File: source/module/index.rst (hunk 3 above)
Comment (seisman) [reply source/module/index.rst:176]: gmtswitch 我感觉设计的也很糟糕，所以也不想加进去。

Comment (core-man) [reply source/module/index.rst:176]: Okay. Remove gmt:gmt5syntax, gmt:gmt_shell_functions.sh and swith? 

img2grd 可以继续放在**主程序/脚本**分类中?

Comment (seisman) [reply source/module/index.rst:176]: 虽然这个用起来是个脚本，但其实当成一个模块也是OK的，我觉得把它放在脚本里有点过于强调它了

Comment (seisman) [reply source/module/index.rst:176]: 像isogmt，本来也是有的，后来也被我删掉了，因为没用

Comment (core-man) [reply source/module/index.rst:176]: `gmt:gmt5syntax` and `gmt:gmt_shell_functions.sh` are removed in e78b1c0.

**Review by seisman — APPROVED** [review]

**Comment by seisman** [comment]
Please work on other categories in separate PRs.

**Pull request merged**